### ISSUE TITLE
[BUG] SemanticsNode tried to set a non-finite rect

### ISSUE THREAD

**Issue by hhoang** [flutter/flutter#121820]
Hello, 
i'm receiving the following error when attempting to animate a button inside a CustomScrollView with slivers.

```======== Exception caught by scheduler library =====================================================
The following assertion was thrown during a scheduler callback:
SemanticsNode#5(Rect.fromLTRB(1.0, 1.0, 392.0, 18.0), tags: [RenderViewport.twoPane], label: "hello", textDirection: ltr) (with SemanticsOwner#0852b) tried to set a non-finite rect.
'package:flutter/src/semantics/semantics.dart':
Failed assertion: line 1605 pos 12: 'value.isFinite'


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.md

When the exception was thrown, this was the stack: 
#2      SemanticsNode.rect= (package:flutter/src/semantics/semantics.dart:1605:12)
#3      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4223:11)
#4      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4235:16)
#5      RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:3159:25)
#6      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1221:16)
#7      RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:518:21)
#8      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:884:13)
#9      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:378:5)
#10     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1175:15)
#11     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1104:9)
#12     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1015:5)
#13     _invoke (dart:ui/hooks.dart:148:13)
#14     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:318:5)
#15     _drawFrame (dart:ui/hooks.dart:115:31)
(elided 2 frames from class _AssertionError)
====================================================================================================
```

Flutter doctor output:

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.2, on macOS 12.6.1 21G217 darwin-arm, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.1)
[✓] VS Code (version 1.74.3)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!

## Steps to Reproduce
I've created a gist here that reproduces the exception:

https://gist.github.com/hhoang/2c54c8b13d9a88d0c74972cdb05869ea

Screen recording of the issue:
https://user-images.githubusercontent.com/482679/222529254-9bc8b02d-539e-4af0-9538-2c9b93c6351a.mov

**Comment by darshankawar** [comment]
Thanks for the report @hhoang 
Running your code sample on latest master gives me same exception as reported. Although if you see the log, it says 
```
`label: "hello", textDirection: ltr) (with SemanticsOwner#0e769) tried to set a non-finite rect.
'package:flutter/src/semantics/semantics.dart':
Failed assertion: line 1712 pos 12: 'value.isFinite'
```
So looking at your code implementation, you are passing `cacheExtent: double.maxFinite,` which seems to be causing the exception. If you pass `cacheExtent: double.minPositive`, then the exception doesn't occur which satisfies the assertion `'value.isFinite'` and see if it helps to serve the purpose.

You may read about the usage and working of `cacheExtent` here for further reference https://api.flutter.dev/flutter/rendering/RenderViewportBase/cacheExtent.html

**Comment by hhoang** [comment]
thanks @darshankawar  setting it to minPositive resolves the exception. However I am confused since `maxFinite` seems to imply that the value is finite, which should also satisfy the assertion `value.isFinite`. 

My concern is that changing this would negatively impact the scrolling behavior of my list.

**Comment by darshankawar** [comment]
Thanks for the feedback. Considering that the error log prompts to file an issue, I'll reopen it and label for team's attention, as, I do see the same assertion failure on latest versions per my earlier comment.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.7.5, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.7.5 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c07f788888 (7 days ago), 2023-02-22 17:52:33 -0600
    • Engine revision 0f359063c4
    • Dart version 2.19.2
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.9.0-1.0.pre.49, on macOS 12.2.1 21D62 darwin-x64,
    locale en-GB)
    • Flutter version 3.9.0-1.0.pre.49 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 95387c6621 (28 hours ago), 2023-03-04 16:59:14 -0800
    • Engine revision de724e2632
    • Dart version 3.0.0 (build 3.0.0-290.0.dev)
    • DevTools version 2.22.2
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by HansMuller** [comment]
I agree that this should probably work. On the other hand, making the cacheExtent essentially infinite torpedoes the advantages of using a customScrollView. It implies that the entire list should be laid out. If that's really whats intended then it would be simpler to just create a SingleChildScrollView with a Column child.

**Comment by plammens** [comment]
I have a similar issue but nowhere have I set `cacheExtent`. I'm working on making a minimal example, but meanwhile is there anything that I should be looking at?

**Comment by chunhtai** [comment]
Is this still an issue? there was a major refactoring around that area of the code
